aip/cloud/25001.md

```diff
@@ -0,0 +1,66 @@
+---
+id: 25001
+state: draft
+created: TBD
+placement:
+  category: gcp
+---
+
+# Empty Objects
+
+Fields of type Message **must not** be empty. Empty messages do not have uniform
+
+## Alternatives
+
+The presence of empty Messages are sometimes used to indicate that a flag is enabled. Alternatives to empty Messages could include a single boolean field within the Message to signify enablement or an Enum rather than an empty Message.
+
+### Add Field to Message
+For example:
+```proto
+message EnableMyFeature {
+}
+```
+Could be modeled instead as:
+
+```proto
+message MyFeature {
+  bool enabled = 1;
+}
+```
+
+### Move Option to Top-Level Enum
+
+For a situation where the presence of an empty Message indicates a certain option is being chosen, an alternative is to choose the option via an enum rather than the presence of a particular object.
+
+For example:
+```proto
+oneof access_model {
+  AccessModelFree free = 1
```

Comment:
would it make sense for this oneof to have unspecified? or some default value, as with any other enum value?

IIRC that works around backwards compatibility issues where the returned enum is not a value that is present in older versions of a client.